### PR TITLE
Add contributing.md

### LINKED ISSUE #480: Document the CLI utilities in `ethereum_spec_tools` in `README.md`
We have a couple neat CLI tools that we should document in case anyone wants to use them.

 - [x] Patch Tool
 - [x] Sync Tool
 - [x] Lint Tool
 - [x] Fork Tool (#478)

### PR DESCRIPTION
**Pull request by gurukamath**
(closes #480 )

### What was wrong?
No documentation on the development tools is currently available

Related to Issue #480 

### How was it fixed?
Created `contributing.md`, moved relevant sections from `README.md` to it and added the missing developer tool documentation.

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+```bash
+ethereum-spec-new-fork --from_fork="Tangerine Whistle" --to_fork="Spurious Dragon" --from_test=EIP150 --to_test=EIP158
+```
+
+The following will have to however, be updated manually
+ 1. The fork number and MAINNET_FORK_BLOCK in __init__.py
+ 2. Any absolute package imports from other forks eg. in trie.py
+ 3. Package names under setup.cfg
+ 4. Add the new fork to the monkey_patch() function in src/ethereum_optimized/__init__.py
+ 5. Adjust the underline in fork/__init__.py
```
Comment (SamWilsn) [inline CONTRIBUTING.md:103]:
```suggestion
 5. Adjust the underline in `fork/__init__.py`
```

**Review by SamWilsn — COMMENTED** [review]
- Should mention that all code must be licensed under CC0.
 - Should we have a section on naming things? One suggestion:
   - Don't use EIP numbers in identifiers, and try to pick descriptive names instead.

**Comment by gurukamath** [comment]
> * Should mention that all code must be licensed under CC0.
> * Should we have a section on naming things? One suggestion:
>   
>   * Don't use EIP numbers in identifiers, and try to pick descriptive names instead.

Added license to Readme. Added naming to the existing spelling section

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
+# Contribution Guidelines
+
+Help is always welcome and there are plenty of options to contribute to EELS.
+
+In particular, we appreciate support in the following areas:
+
+- Reporting issues
+- Fixing and responding to [issues](https://github.com/ethereum/execution-specs/issues), especially those tagged as [good first issue](https://github.com/ethereum/execution-specs/labels/good%20first%20issue) which are meant as introductory issues for external contributors.
```
[the remark this answers is not in the document]
Comment (gurukamath) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+### Development
+
+Running the tests necessary to merge into the repository requires:
+
+ * Python 3.8.x (not 3.9 or later), and
+ * [PyPy 7.3.x](https://www.pypy.org/).
+ * `geth` installed and present in `$PATH`
+
+These version ranges are necessary because, at the time of writing, PyPy is only compatible with Python 3.7.
```
[the remark this answers is not in the document]
Comment (gurukamath) [reply CONTRIBUTING.md]: Updated. Please check.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+### Spelling
+
+Attempt to use descriptive English words (or _very common_ abbreviations) in documentation and identifiers. If necessary, there is a custom dictionary `whitelist.txt`.
+
+
+### Development
+
+Running the tests necessary to merge into the repository requires:
+
+ * Python 3.8.x (not 3.9 or later), and
```
[the remark this answers is not in the document]
Comment (gurukamath) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
+# Contribution Guidelines
+
+Help is always welcome and there are plenty of options to contribute to EELS.
```
[the remark this answers is not in the document]
Comment (gurukamath) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+Below is an example.
+
+```bash
+pytest tests/frontier/test_state_transition.py -k 'test_general_state_tests_new' --evm-trace
+```
+
+
+## CLI Utilities `ethereum_spec_tools`
+
+The execution specs repository has various CLI utilities that can help in the development process.
```
[the remark this answers is not in the document]
Comment (gurukamath) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+ * to_test (Optional): Name of the to fork within the test fixtures in case it is different from fork name Eg - "EIP158"
+
+As an example, if one wants to create baseline code for the `Spurious Dragon` fork from the `Tangerine Whistle` one
+
+```bash
+ethereum-spec-new-fork --from_fork="Tangerine Whistle" --to_fork="Spurious Dragon" --from_test=EIP150 --to_test=EIP158
+```
+
+The following will have to however, be updated manually
+ 1. The fork number and MAINNET_FORK_BLOCK in __init__.py
```
[the remark this answers is not in the document]
Comment (gurukamath) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+ 2. Any absolute package imports from other forks eg. in trie.py
```
[the remark this answers is not in the document]
Comment (gurukamath) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+ 3. Package names under setup.cfg
```
[the remark this answers is not in the document]
Comment (gurukamath) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+ 4. Add the new fork to the monkey_patch() function in src/ethereum_optimized/__init__.py
```
[the remark this answers is not in the document]
Comment (gurukamath) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md (hunk 1 above)
Comment (gurukamath) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+Help is always welcome and there are plenty of options to contribute to EELS.
+
+In particular, we appreciate support in the following areas:
+
+- Reporting issues
+- Fixing and responding to [issues](https://github.com/ethereum/execution-specs/issues), especially those tagged as [good first issue](https://github.com/ethereum/execution-specs/labels/good%20first%20issue) which are meant as introductory issues for external contributors.
+- Improving the documentation.
+
+
+For details about the Execution Specifications usage and building, please refer the [README](https://github.com/ethereum/execution-specs/blob/master/README.md#usage)
```
[the remark this answers is not in the document]
Comment (gurukamath) [reply CONTRIBUTING.md]: Updated

**Pull request merged**